### ISSUE TITLE
Can the order of MAJOR.MINOR.BUILD.PATCH be changed??

### ISSUE THREAD

**Issue by dnrelectronica** [Asmodai/verbuild#18]
Hi, 

I have a question regarding the order of the four elements which compose the final version number. 
In the readme file i understand the order is as follows:

MAJOR.MINOR.BUILD.PATCH

I don't like the BUILD nr at the third position there. I think it should be at the end and PATCH should be third like this:

MAJOR.MINOR.PATCH.BUILD

This is the semantic versioning format.
https://semver.org/#:~:text=PATCH%2C%20increment%20the%3A,make%20backwards%20compatible%20bug%20fixes.

What is the reason you choose the BUILD to be at the third position? 

In my opinion the BUILD is not so important, but the PATCH is!

I'd like to have a version like: v1.0.2 - build 3156

How can i change this order in the verbuild tool?? change code somewhere, and recompile? 

I hope you can help me out. 

Kind Regards, 

Corné

**Comment by Asmodai** [comment]
As of right now, Verbuild knows nothing about semantic versioning's approach, as it actually predates semver by a while (the versioning system is based on that used by vendors such as Microsoft, for example.)

The theory there goes that major.minor.build is more important than major.minor.patch as for software such as Windows, a patch is applied to a build.  This is why Verbuild does it this way.

It would be possible to change the format to one you require by editing `Transform_C.cpp` to write out whatever format you want, at least for now.

I am in the process of overhauling Verbuild to include Lua, which will make this task easier in the long run, as the generation aspect of it will no longer be hard-coded into the binary :)

I'm also going to add explicit semver support, because it sounds like that would actually be a useful addition to have.

**Comment by Asmodai** [comment]
I've hacked together some rather simplistic support for SemVer for testing.

Sources at https://github.com/Asmodai/verbuild/tree/semver
Release (and Windows 64-bit binary) available at https://github.com/Asmodai/verbuild/releases/tag/2.1.615.0

**Caveats**: 

- This hack makes no attempt to validate metadata using the semver BNF grammar,
- This hack only writes semver metadata to `VERSION_STRING`

**Example**:
```
$ ./verbuild.exe -s true -m "alpha" -t c -g basic -o test.h
[OK] Version incremented to: 0.1.0-alpha+1
$ cat test.h
/*
 * ----------------------------------------------------
 * Automatically generated by VerBuild 2.1.615.0.
 * Do not edit by hand.
 * ----------------------------------------------------
 */

#pragma once

#ifndef __VersionInfo_Header__
#define __VersionInfo_Header__

#define VERSION_MAJOR      0
#define VERSION_MINOR      1
#define VERSION_BUILD      1
#define VERSION_PATCH      0

#define VERSION_BASE_YEAR  2020
#define VERSION_DATE       "2020-Jul-04"
#define VERSION_TIME       "15:37:22"
#define VERSION_STRING     "0.1.0-alpha+1"

#endif // !__VersionInfo_Header__
$ ./verbuild.exe -s true -m "alpha" -t c -g basic -o test.h
[OK] Version incremented to: 0.1.0-alpha+2
$ 
```

**Comment by dnrelectronica** [comment]
Hi @Asmodai,

Is the SemVer feature already in the master repo? 

Kind regards,
Corne Doggen

**Comment by Asmodai** [comment]
No, it's in the `semver` branch, as it's currently a nasty hack and will be replaced by something better soon-ish.
